scripts/download/tools.ts

```diff
@@ -97,133 +127,148 @@ async function findChecksum(checksumURL, executableName) {
+  const dockerComposeURLBase = `https://github.com/docker/compose/releases/download/${ version }`;
   const dockerComposeCPU = process.env.M1 ? 'aarch64' : 'x86_64';
-  const dockerComposeExecutable = exeName(`docker-compose-${ kubePlatform }-${ dockerComposeCPU }`);
+  const dockerComposeExecutable = exeName(context, `docker-compose-${ context.kubePlatform }-${ dockerComposeCPU }`);
   const dockerComposeURL = `${ dockerComposeURLBase }/${ dockerComposeExecutable }`;
-  const dockerComposePath = path.join(binDir, exeName('docker-compose'));
+  const dockerComposePath = path.join(context.binDir, exeName(context, 'docker-compose'));
   const dockerComposeSHA = await findChecksum(`${ dockerComposeURL }.sha256`, dockerComposeExecutable);
 
   await download(dockerComposeURL, dockerComposePath, { expectedChecksum: dockerComposeSHA });
+}
 
+async function downloadTrivy(context: DownloadContext, version: string): Promise<void> {
   // Download Trivy
-  // Always run this in the VM, so download the *LINUX* version into binDir
+  // Always run this in the VM, so download the *LINUX* version into internalDir
   // and move it over to the wsl/lima partition at runtime.
   // This will be needed when RD is ported to linux as well, because there might not be
   // an image client running on the host.
   // Sample URLs:
   // https://github.com/aquasecurity/trivy/releases/download/v0.18.3/trivy_0.18.3_checksums.txt
   // https://github.com/aquasecurity/trivy/releases/download/v0.18.3/trivy_0.18.3_macOS-64bit.tar.gz
 
-  const trivyVersionWithV = 'v0.30.0';
+  const versionWithV = `v${ version }`;
   const trivyURLBase = `https://github.com/aquasecurity/trivy/releases`;
-  const trivyVersion = trivyVersionWithV.replace(/^v/, '');
-  const trivyOS = cpu === 'amd64' ? 'Linux-64bit' : 'Linux-ARM64';
-  const trivyBasename = `trivy_${ trivyVersion }_${ trivyOS }`;
-  const trivyURL = `${ trivyURLBase }/download/${ trivyVersionWithV }/${ trivyBasename }.tar.gz`;
-  const trivySHA = await findChecksum(`${ trivyURLBase }/download/${ trivyVersionWithV }/trivy_${ trivyVersion }_checksums.txt`, `${ trivyBasename }.tar.gz`);
-
-  // Grab a linux executable and put it in the linux/internal dir, which will
-  // probably need to be created
-  const linuxInternalDir = path.join(process.cwd(), 'resources', 'linux', 'internal');
-  const trivyPath = path.join(linuxInternalDir, 'trivy');
-
-  await fs.promises.mkdir(linuxInternalDir, { recursive: true });
+  const trivyOS = process.env.M1 ? 'Linux-ARM64' : 'Linux-64bit';
+  const trivyBasename = `trivy_${ version }_${ trivyOS }`;
+  const trivyURL = `${ trivyURLBase }/download/${ versionWithV }/${ trivyBasename }.tar.gz`;
+  const checksumURL = `${ trivyURLBase }/download/${ versionWithV }/trivy_${ version }_checksums.txt`;
+  const trivySHA = await findChecksum(checksumURL, `${ trivyBasename }.tar.gz`);
+  const trivyPath = path.join(context.internalDir, 'trivy');
```

Comment:
Is `context.internalDir` correct? Isn't that `resources/darwin/internal` (on mac), etc.?

Reply:
Nice catch, you're right. It looks like in `main`, it puts trivy into `resources/linux/internal`, always. Same thing with the guest agent. I'll fix this.